src/daemon/ca/certauth.rs

```diff
@@ -207,113 +232,168 @@ impl Aggregate for CertAuth {
+                self.parents.insert(parent, contact);
             }
-            EvtDet::ParentRemoved(handle, _deltas) => {
-                self.parents.remove(&handle);
-                self.resources.retain(|_, rc| rc.parent_handle() != &handle);
+            CaEvtDet::ParentRemoved { parent } => {
+                self.parents.remove(&parent);
+                self.resources.retain(|_, rc| rc.parent_handle() != &parent);
             }
 
-            EvtDet::ResourceClassAdded(name, rc) => {
+            CaEvtDet::ResourceClassAdded {
+                resource_class_name,
+                parent,
+                parent_resource_class_name,
+                pending_key,
+            } => {
                 self.next_class_name += 1;
-                self.resources.insert(name, rc);
+                let ns = resource_class_name.to_string();
+                let rc = ResourceClass::create(
+                    resource_class_name.clone(),
+                    ns,
+                    parent,
+                    parent_resource_class_name,
+                    pending_key,
+                );
+                self.resources.insert(resource_class_name, rc);
             }
-            EvtDet::ResourceClassRemoved(name, _delta, _parent, _revocations) => {
-                self.resources.remove(&name);
+            CaEvtDet::ResourceClassRemoved {
+                resource_class_name, ..
+            } => {
+                self.resources.remove(&resource_class_name);
             }
-            EvtDet::CertificateRequested(name, req, status) => {
-                self.resources.get_mut(&name).unwrap().add_request(status, req);
+            CaEvtDet::CertificateRequested {
+                resource_class_name,
+                req,
+                ki,
+            } => {
+                self.resources
+                    .get_mut(&resource_class_name)
+                    .unwrap()
+                    .add_request(ki, req);
             }
-            EvtDet::CertificateReceived(class_name, key_id, cert) => {
-                self.resources.get_mut(&class_name).unwrap().received_cert(key_id, cert);
+            CaEvtDet::CertificateReceived {
+                resource_class_name,
+                ki,
+                rcvd_cert,
+            } => {
+                self.resources
+                    .get_mut(&resource_class_name)
+                    .unwrap()
+                    .received_cert(ki, rcvd_cert);
             }
 
             //-----------------------------------------------------------------------
             // Key Life Cycle
             //-----------------------------------------------------------------------
-            EvtDet::KeyRollPendingKeyAdded(class_name, key_id) => {
-                self.resources.get_mut(&class_name).unwrap().pending_key_added(key_id);
+            CaEvtDet::KeyRollPendingKeyAdded {
+                resource_class_name,
+                pending_key,
```

Comment:
should `_key` here be `key_id` as it was before, as a key is perhaps different than a key ID?